vms/platformvm/txs/executor/proposal_tx_preference.go

```diff
@@ -0,0 +1,179 @@
+	return ErrWrongTxType
+}
+
+func (*ProposalTxPreference) BaseTx(*txs.BaseTx) error {
+	return ErrWrongTxType
+}
+
+func (*ProposalTxPreference) AddValidatorTx(*txs.AddValidatorTx) error {
+	return ErrWrongTxType
+}
+
+func (*ProposalTxPreference) AddSubnetValidatorTx(*txs.AddSubnetValidatorTx) error {
+	return ErrWrongTxType
+}
+
+func (*ProposalTxPreference) AddDelegatorTx(*txs.AddDelegatorTx) error {
+	return ErrWrongTxType
+}
+
+func (*ProposalTxPreference) AdvanceTimeTx(*txs.AdvanceTimeTx) error {
+	return ErrWrongTxType
+}
+
+func (e *ProposalTxPreference) RewardValidatorTx(tx *txs.RewardValidatorTx) error {
+	stakerTx, _, err := e.State.GetTx(tx.TxID)
+	if err == database.ErrNotFound {
+		// This can happen if this transaction is attempting to reward a
+		// validator that hasn't been confirmed.
+		e.PrefersCommit = true
+		return nil
+	}
+	if err != nil {
+		// GetTx can only return [ErrNotFound], or an unexpected error like a
+		// parsing error or internal DB error. For anything other than
+		// [ErrNotFound] the block can just be dropped.
+		return fmt.Errorf("%w %s: %w",
+			ErrMissingStakerTx,
+			tx.TxID,
+			err,
+		)
+	}
+
+	staker, ok := stakerTx.Unsigned.(txs.Staker)
+	if !ok {
+		// Because this transaction isn't guaranteed to have been verified yet,
+		// it's possible that a malicious node issued this transaction into a
+		// block that will fail verification in the future.
+		return fmt.Errorf("%w %s: %T",
+			ErrUnexpectedStakerTransactionType,
+			tx.TxID,
+			stakerTx.Unsigned,
+		)
+	}
+
+	// retrieve primaryNetworkValidator before possibly removing it.
+	nodeID := staker.NodeID()
+	primaryNetworkValidator, err := e.State.GetCurrentValidator(
+		constants.PrimaryNetworkID,
+		nodeID,
+	)
+	if err != nil {
+		// If this transaction is included into an invalid block where the
+		// staker has already been removed, we can just drop it.
+		return fmt.Errorf("%w %s: %w",
+			ErrStakerWithoutPrimaryNetworkValidator,
+			nodeID,
+			err,
+		)
+	}
+
+	expectedUptimePercentage := e.PrimaryUptimePercentage
+	if subnetID := staker.SubnetID(); subnetID != constants.PrimaryNetworkID {
+		transformSubnet, err := GetTransformSubnetTx(e.State, subnetID)
+		if err != nil {
+			// If the subnet hasn't been transformed yet, the tx we are removing
+			// isn't for a permissionless subnet. So, it's removal here is
+			// invalid.
+			return fmt.Errorf("%w %s: %w",
+				ErrMissingSubnetTransformation,
+				subnetID,
+				err,
+			)
+		}
+
+		expectedUptimePercentage = float64(transformSubnet.UptimeRequirement) / reward.PercentDenominator
+	}
+
+	uptime, err := e.Uptimes.CalculateUptimePercentFrom(
+		nodeID,
+		constants.PrimaryNetworkID,
+		primaryNetworkValidator.StartTime,
+	)
+	if err != nil {
+		// If this transaction is included into an invalid block where the
+		// staker has already been removed, we can just drop it.
+		return fmt.Errorf("%w %s: %w",
+			ErrCalculatingUptime,
+			nodeID,
+			err,
+		)
```

Comment:
Because we are guaranteed that the node is a current validator of the primary network at this point, I don't think this can ever cause a FATAL.